test/functional/feature_anchors.py

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env python3
+# Copyright (c) 2020 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test Anchors functionality"""
+
+from test_framework.p2p import P2PInterface
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import assert_equal
+
+import os
+
+def check_node_connections(*, node, num_in, num_out):
+    info = node.getnetworkinfo()
+    assert_equal(info["connections_in"], num_in)
+    assert_equal(info["connections_out"], num_out)
+
+
+class AnchorsTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.num_nodes = 1
+
+    def setup_network(self):
+        self.setup_nodes()
+
+    def run_test(self):
+        self.log.info("Add 2 block-relay-only connections to node 0")
+        for i in range(2):
+            self.log.info(f"block-relay-only: {i}")
+            self.nodes[0].add_outbound_p2p_connection(P2PInterface(), p2p_idx=i, connection_type="block-relay-only")
+
+        self.log.info("Add 5 inbound connections to node 0")
+        for i in range(5):
+            self.log.info(f"inbound: {i}")
+            self.nodes[0].add_p2p_connection(P2PInterface())
+
+        self.log.info("Check node 0 connections")
+        check_node_connections(node=self.nodes[0], num_in=5, num_out=2)
+
+        # 127.0.0.1
+        ip = '7f000001'
+
+        # Since the ip is always 127.0.0.1 for this case, we store only the port to identify the peers
+        block_relay_nodes_port = []
+        inbound_nodes_port = []
+        for p in self.nodes[0].getpeerinfo():
+            addr_split = p['addr'].split(':')
+            if p['connection_type'] == 'block-relay-only':
+                block_relay_nodes_port.append(hex(int(addr_split[1]))[2:])
+            else:
+                inbound_nodes_port.append(hex(int(addr_split[1]))[2:])
+
+        self.log.info("Stop node 0")
+        self.stop_node(0)
+
+        self.log.info("Check anchors.dat file in the node directory")
+        assert os.path.exists(os.path.join(self.nodes[0].datadir + '/regtest', 'anchors.dat'))
+
+        # It should contains only the block-relay-only addresses
+        self.log.info("Check the addresses in anchors.dat")
+        for line in open(os.path.join(self.nodes[0].datadir + '/regtest', 'anchors.dat'), 'rb'):
```

Comment:
```suggestion
        for line in open(os.path.join(self.nodes[0].datadir + '/regtest', 'anchors.dat'), 'rb', encoding="utf8"):
```

Reply:
I think it won't work because it's binary mode.

Reply:
Anyway, all of the linters must pass.

Reply:
Tested ACK. I'm pretty sure It's a false-flag.
- OS: Ubuntu 20.10

### Without `encoding` - passes
![Screenshot from 2021-03-02 14-07-31](https://user-images.githubusercontent.com/7918067/109687248-d32e4080-7b61-11eb-86d5-0e73cb6de557.png)

### With `encoding` - fail
![Screenshot from 2021-03-02 14-15-21](https://user-images.githubusercontent.com/7918067/109687265-d6c1c780-7b61-11eb-89db-9e8c930ea052.png)

---

I also have checked other tests that are reading binaries. Such as [p2p_message_capture.py (line 37)](https://github.com/bitcoin/bitcoin/blob/b9f41df1ead4b6a83a51fc41966b111c8459c313/test/functional/p2p_message_capture.py#L37) or [rpc_dumptxoutset.py (line 40)](https://github.com/bitcoin/bitcoin/blob/b9f41df1ead4b6a83a51fc41966b111c8459c313/test/functional/rpc_dumptxoutset.py#L40). None have the encoding parameter, which makes sense.